airflow/contrib/hooks/gcp_dlp_hook.py

```diff
@@ -0,0 +1,1362 @@
+            to complete. Note that if retry is specified, the timeout applies to each
+            individual attempt.
+        :type timeout: float
+        :param metadata: (Optional) Additional metadata that is provided to the method.
+        :type metadata: sequence[tuple[str, str]]]
+        """
+
+        client = self.get_conn()
+
+        if not name:
+            raise AirflowException(
+                "Please provide the name of the DlpJob resource to be cancelled."
+            )
+
+        client.cancel_dlp_job(
+            name=name, retry=retry, timeout=timeout, metadata=metadata
+        )
+
+    @GoogleCloudBaseHook.catch_http_exception
+    def create_deidentify_template(
+        self,
+        parent,
+        deidentify_template=None,
+        template_id=None,
+        retry=None,
+        timeout=None,
+        metadata=None,
+    ):
+        """
+        Creates a DeidentifyTemplate for re-using frequently used configuration for
+        de-identifying content, images, and storage.
+
+        :param parent: The parent resource name.
+        :type parent: str
+        :param deidentify_template: (Optional) The DeidentifyTemplate to create.
+        :type deidentify_template: dict or google.cloud.dlp_v2.types.DeidentifyTemplate
+        :param template_id: (Optional) The template ID.
+        :type template_id: str
+        :param retry: (Optional) A retry object used to retry requests.
+            If None is specified, requests will not be retried.
+        :type retry: google.api_core.retry.Retry
+        :param timeout: (Optional) The amount of time, in seconds, to wait for the request
+            to complete. Note that if retry is specified, the timeout applies to each
+            individual attempt.
+        :type timeout: float
+        :param metadata: (Optional) Additional metadata that is provided to the method.
+        :type metadata: sequence[tuple[str, str]]]
+        :rtype: google.cloud.dlp_v2.types.DeidentifyTemplate
+        """
+
+        client = self.get_conn()
+
+        if not parent:
+            raise AirflowException("Please provide the parent resource name.")
+
+        return client.create_deidentify_template(
+            parent=parent,
+            deidentify_template=deidentify_template,
+            template_id=template_id,
+            retry=retry,
+            timeout=timeout,
+            metadata=metadata,
+        )
+
+    @GoogleCloudBaseHook.catch_http_exception
+    def create_dlp_job(
+        self,
+        parent,
+        inspect_job=None,
+        risk_job=None,
+        job_id=None,
+        retry=None,
+        timeout=None,
+        metadata=None,
+        wait_until_finished=True,
+        polling_interval_in_seconds=60
+    ):
+        """
+        Creates a new job to inspect storage or calculate risk metrics.
+
+        :param parent: The parent resource name.
+        :type parent: str
```

Comment:
This contains the project ID. I think it's worth adding support for setting the project ID by configuring the connection
Please look at: `@GoogleCloudBaseHook.fallback_to_default_project_id`